github-upload-secrets

```diff
@@ -77,20 +76,30 @@ def generate_ssh_key(pubkey):
                                    stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         private, public = process.communicate(b'y\n')
 
-    return public, encrypt(pubkey["key"], private)
+    return public, private
 
 
-def read_and_encrypt(filename, pubkey):
-    with open(filename, 'rb') as f:
-        return encrypt(pubkey["key"], f.read().strip())
+def upload_secrets(api, repo, env, secrets, opts):
+    env_path = f"/repos/{repo}/environments/{env}"
```

Comment:
I think it was you who talked me into using proper URL encoding, so that envs with funny names would work. No veto, but it's not so complicated to do it right..

Reply:
The reason it's a bit weird in this case is because we now inject the environment as part of a org/repo/env/secret quad, and the main thing that you'd escape from URLs is the `/` itself, which is no longer possible to inject.

Also: this quad gets subjected to a regexp specifically to check that it's not exotic...

For what it's worth, I intend to kill off `--repository` and `--env` fairly soon, at which point this is all going to be substantially simplified.

Reply:
Ack, good enough for sure. Resolving.